NitroxClient/GameLogic/Bases/Spawning/BasePiece/BasePieceSpawnPrioritizer.cs

```diff
@@ -77,5 +81,54 @@ private int ComputeBasePiecePriority(BasePieceData basePiece)
 
             return piecesToPriority.TryGetValue(techType, out int position) ? position : int.MaxValue;
         }
+
+        private List<BasePieceData> OrderByHeight(IEnumerable<BasePieceData> orderedList, TechType techType)
```

Comment:
I don't know for sure how to improve this piece of code, but this should probably tested